design/layer2-prefer-leader.md

```diff
@@ -0,0 +1,103 @@
+# Configuring Preferred Node To Be Layer2 Leader
+
+## Summary
+
+The purpose of the ehancement is to support configuring preferred node to be the leader for L2 mode.
+
+With this enhancement, the leader election will be artificially controllable.
+
+## Motivation
+
+MetalLB’s layer2 mode is simliar to `Keepalived`(https://metallb.universe.tf/concepts/layer2/#comparison-to-keepalived) which using `VRRP`.
+
+This enhancement implement functionality similar to [Priority and Preemption](https://www.cisco.com/assets/sol/sb/Switches_Emulators_v2_3_5_xx/help/350_550/index.html#page/tesla_350_550_olh/ts_vrrp_18_09.html) configuration of `VRRP`.
+
+### Goals
+
+- In L2 mode, being able to specify a node as the preferred leader for a service.
+- When a node is configured as the preferred leader, it replaces the current leader.
+
+### Non-Goals
+
+- Change the current leader mechanism election using memberlist.
+- Change the data models of Metallb's CRDs.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+When using Metallb's Layer2 mode, user wants to specify one node as the leader for a service and let all traffic for this service's IP goes to this sepcified node.
+
+#### Story 2
+
+When using Metallb's Layer2 mode, the current leader node requires maintenance, user want to manually switchover to a sepcified node.
```

Comment:
How this would be achieved with this feature? By changing the leader manually?
I the node is not available metallb should automatically detect it and move the ip to another node. If we want to exclude a node, this can be done either via the node selectors we now have and / or playing with the speaker node selector.

Reply:
Maintenance doesn't mean the node is not available. For example, if a troubleshooting step requires, we may need the traffic for a service only passing through certain node to make capturing possible. As far as I understand, the nodeselector would affect all services served by the L2Advertisement, which may affect a much larger scope.

Reply:
If troubleshooting is needed, you can have (or it will be possible in the next release) an ipaddresspool tied to a given service with a label selector, tie the advertisement to that pool only and so affect only a single service.